Updated Upstream (Bukkit/CraftBukkit/Spigot)

Machine-Maker opened the pull request:
Upstream has released updates that appear to apply and compile correctly.
This update has not been tested by PaperMC and as with ANY update, please do your own testing

Bukkit Changes:
146a7e4b SPIGOT-5345: Add automatic library support

CraftBukkit Changes:
b1064c69 Remove sisu annotation processor from jar
32e40866 SPIGOT-6189: Persistent data disappears when calling setFacingDirection on an item frame
d189f78b #&#x2060;827: Trigger vanilla dimension advancements in non-main worlds
5bbb4a65 Add plumbing for automatic library support

Spigot Changes:
9fb885e8 Rebuild patches

Proximyst reviewed (approved):
Looks fine to me. Would like @PaperMC/development-team to also do a proper read through to make sure I didn't miss anything.

MiniDigger commented on Spigot-Server-Patches/0348-Fix-sounds-when-item-frames-are-modified-MC-123450.patch:
```diff
@@ -18,16 +18,3 @@ index 584f64946821092afab57b9409bc249403bc16e7..43152a6c70c9433d627a58051101530d
- 
-         EntityItemFrame frame = new EntityItemFrame(world, position, direction);
--        frame.setItem(item);
-+        frame.setItem(item, true, false); // Paper - fix itemframe sound
```
just to double check, this was dropped intentionally, right?

Proximyst replied:
Believe that was fixed upstream:
https://hub.spigotmc.org/stash/projects/SPIGOT/repos/craftbukkit/commits/32e4086604f6df169869172175a194c41422de38#src/main/java/org/bukkit/craftbukkit/entity/CraftItemFrame.java

Machine-Maker committed:
Updated Upstream (Bukkit/CraftBukkit/Spigot)

Upstream has released updates that appear to apply and compile correctly.
This update has not been tested by PaperMC and as with ANY update, please do your own testing

Bukkit Changes:
146a7e4b SPIGOT-5345: Add automatic library support

CraftBukkit Changes:
b1064c69 Remove sisu annotation processor from jar
32e40866 SPIGOT-6189: Persistent data disappears when calling setFacingDirection on an item frame
d189f78b #827: Trigger vanilla dimension advancements in non-main worlds
5bbb4a65 Add plumbing for automatic library support

Spigot Changes:
9fb885e8 Rebuild patches
